src/main/java/com/microsoft/cse/helium/app/services/keyvault/KeyVaultService.java

```diff
@@ -95,23 +98,53 @@ public String getKeyVaultUri(){
         return kvUri;
     }
 
-    public String getSecret(String secretName){
-        String returnValue="";
-        
-        String kvUri = getKeyVaultUri();
-
-        SecretBundle secret = _keyVaultClient.getSecret(kvUri, secretName);
-        returnValue = secret.value();
-
-        return returnValue;
+    public Mono<String> getSecret(String secretName){
```

Comment:
I am assuming that because of the way Keyvault is instantiated, it will be likely be a bean i.e. instantiated only once per process and then cached. We could likely take the sync hit for that. Any particular reason converting it to mono when the underlying access is sync only? Making it Mono would hide and make it feel like it is sync when it isn't truly.

Reply:
The underlying access is using the async API. Take look and you'll see it is async all the way through.